www/main/settings/utils/theme-bg-uploader.js

```diff
@@ -51,29 +52,39 @@ export const uploadImage = async (evt) => {
   } catch (error) {
     errorAlert('Unable to create folder');
   }
+  return new Promise((resolve, reject) => {
+    try {
+      const filePath = evt.target.files[0].path;
+      const newPath = path.resolve(userBackgroundsPath, evt.target.files[0].name);
+      // eslint-disable-next-line no-param-reassign
+      evt.target.value = '';
 
-  try {
-    const filePath = evt.target.files[0].path;
-    // eslint-disable-next-line no-param-reassign
-    evt.target.value = '';
-
-    if (imageCheck(filePath)) {
-      const files = await imagemin([filePath], userBackgroundsPath);
-      if (files) {
-        const fileUrl = new URL(`file:///${files[0].path}`).href;
-        const customThemeObj = {
-          type: 'custom',
-          url: `${fileUrl}`.replace(/(%20)/g, '\\ '),
-        };
+      if (imageCheck(filePath)) {
+        sharp(filePath)
+          .jpeg({ mozjpeg: true })
+          .toFile(newPath, (err) => {
+            if (err) {
+              errorAlert('Unable to save file');
+              reject();
+            } else {
+              const customThemeObj = {
+                type: 'custom',
+                url: url.pathToFileURL(newPath),
+              };
 
-        store.setUserPref('app.themebg', customThemeObj);
-        GlobalState.getActions().userSettings.setThemeBg(customThemeObj);
-        global.core.platformMethod('updateSettings');
+              store.setUserPref('app.themebg', customThemeObj);
+              GlobalState.getActions().userSettings.setThemeBg(customThemeObj);
+              global.core.platformMethod('updateSettings');
+              resolve();
+            }
+          });
+      } else {
+        errorAlert('File must be in .png or .jpg format');
+        reject();
       }
-    } else {
-      errorAlert('File must be in .png or .jpg format');
+    } catch (error) {
+      errorAlert('Unknown error occured');
```

Comment:
I think this should come from i18n json file

Reply:
Done.